pyinfra/operations/systemd.py

```diff
@@ -10,17 +10,43 @@
 from .util.service import handle_service_control
 
 
+def _make_systemctl_cmd(user_mode=False, machine=None, user_name=None):
```

Comment:
To avoid duplicating these functions this should be imported from the fact module alongside the facts.

Reply:
Totally agreed. Was a little unhappy I had to send this with the same thing defined in two places.

I attempted the other way around, with a systemd.py in the operations/util directory which I then tried to import into facts/systemd.py; this did not work. No syntax error so valid python but many tests failed which I suspect has to do with how things get loaded internally in pyinfra and I'm lacking specific knowledge about that.

I will try it as you suggest.

Reply:
Updated in [e64cd51](https://github.com/Fizzadar/pyinfra/pull/770/commits/e64cd51ab42bf7481567aa1cbf3593aa076101f0)

Reply:
Interesting the utils approach should work as well, hard to know without seeing the error! I think this way also works, it's hard to say which is the best approach for these things really!

Side thought: maybe facts + operations should live alongside each other rather than divided into two modules.

Reply:
The function is not used in any other operation or fact definitions besides systemd so I think the current approach is understandable.

> Side thought: maybe facts + operations should live alongside each other rather than divided into two modules.

At a minimum having a higher level location for shared utility functions would be nice since apparently there could be some shared logic. The way facts and operations work is quite distinct so I don't know if I'd treat them as equals exactly.